lib/cartopy/io/ogc_clients.py

```diff
@@ -232,8 +232,8 @@ def _image_and_extent(self, wms_proj, wms_srs, wms_extent, output_proj,
         return LocatedImage(wms_image, extent)
 
     def fetch_raster(self, projection, extent, target_resolution):
-        service = self.service
         min_x, max_x, min_y, max_y = extent
+        target_resolution = [np.ceil(val) for val in target_resolution]
```

Comment:
Is taking the ceiling the correct choice. I'm not familiar with this area of cartopy, but naively one might expect rounding.